diag_manager/fms_diag_field_object.F90

```diff
@@ -0,0 +1,965 @@
+!!!       & get_ticks_per_second
+
+use platform_mod
+use iso_c_binding
+
+implicit none
+
+!> \brief Object that holds all variable information
+type fmsDiagField_type
+#ifdef use_yaml
+     type (diagYamlFilesVar_type), allocatable, dimension(:) :: diag_field !< info from diag_table for this variable
+     integer,                      allocatable, dimension(:) :: file_ids   !< Ids of the FMS_diag_files the variable
+                                                                           !! belongs to
+#endif
+     integer, allocatable, private                    :: diag_id           !< unique id for varable
+     type(fmsDiagAttribute_type), allocatable         :: attributes(:)     !< attributes for the variable
+     integer,              private                    :: num_attributes    !< Number of attributes currently added
+     logical, allocatable, private                    :: static            !< true if this is a static var
+     logical, allocatable, private                    :: registered        !< true when registered
+     logical, allocatable, private                    :: mask_variant      !< If there is a mask variant
+     logical, allocatable, private                    :: do_not_log        !< .true. if no need to log the diag_field
+     logical, allocatable, private                    :: local             !< If the output is local
+     TYPE(time_type), private                         :: init_time         !< The initial time
+     integer,          allocatable, private           :: vartype           !< the type of varaible
+     character(len=:), allocatable, private           :: varname           !< the name of the variable
+     character(len=:), allocatable, private           :: longname          !< longname of the variable
+     character(len=:), allocatable, private           :: standname         !< standard name of the variable
+     character(len=:), allocatable, private           :: units             !< the units
+     character(len=:), allocatable, private           :: modname           !< the module
+     character(len=:), allocatable, private           :: realm             !< String to set as the value
+                                                                           !! to the modeling_realm attribute
+     character(len=:), allocatable, private           :: interp_method     !< The interp method to be used
+                                                            !! when regridding the field in post-processing.
+                                                            !! Valid options are "conserve_order1",
+                                                            !! "conserve_order2", and "none".
+     integer, allocatable, dimension(:), private      :: frequency         !< specifies the frequency
+     integer, allocatable, dimension(:), private      :: output_units
+     integer, allocatable, private                    :: t
```

Comment:
Missing comments :-)

Reply:
Probably because these variables should be deleted